src/free_heap_sensor.rs

```diff
@@ -31,7 +34,8 @@ impl FreeHeapSensor {
     }
 }
 
-impl Sensor for FreeHeapSensor {
+impl Sensor for FreeHeapSensor {}
+impl Readings for FreeHeapSensor {
```

Comment:
Pulled this locally, I think `get_generic_readings` needs to be `&mut self` here and elsewhere.